model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -429,3 +442,46 @@ private data class NodeKey(
 
 private val DependencyReference.key: NodeKey
     get() = NodeKey(pkg, fragment)
+
+private enum class NodeColor { WHITE, GRAY, BLACK }
+
+/**
+ * A depth-first-search (DFS)-based implementation which breaks all cycles in O(V + E).
+ * Finding a minimal solution is NP-complete.
+ */
+internal fun breakAllCycles(edges: Collection<Pair<Int, Int>>): Set<Pair<Int, Int>> {
```

Comment:
Nit: Does the "All" as part of the function name add any value? Could it be just "breakCycles"?